test-zone-data/MethodsV2/README.md

```diff
@@ -0,0 +1,1286 @@
+# Implementation of test scenarios for MethodsV2
+
+
+## Table of contents
+
+* [Background](#background)
+* [Root hint for scenarios](#root-hint-for-scenarios)
+* [Undelegated data for scenarios](#undelegated-data-for-scenarios)
+* [Data type in method output](#data-type-in-method-output)
+* [Scenarios](#scenarios)
+  * [GOOD-1](#good-1)
+  * [GOOD-2](#good-2)
+  * [GOOD-3](#good-3)
+  * [GOOD-UNDEL-1](#good-undel-1)
+  * [GOOD-UNDEL-2](#good-undel-2)
+
+## Background
+
+See [Specification of test scenarios for MethodsV2] for the specification of
+data and the definition of the scenarios.
+
+## Root hint for scenarios
+
+For all scenarios the root hint should be
+
+```
+ns1.      A      127.1.0.1
+ns1.      AAAA   fda1:b2:c3::127:1:0:1
+
+ns2.      A      127.1.0.2
+ns2.      AAAA   fda1:b2:c3::127:1:0:2
+```
+
+## Undelegated data for scenarios
+
+If the scenario dependends on undelegated data it is provided with the
+scenario below.
+* The data is provided as unordered and unique pairs of name server name and IP
+  address.
+  * Name and address are separated by "/".
+  * IP address can be empty and then there is no "/".
+* The name server name can be repeated with different IP addresses (IPv4 or
+  IPv6) providing multiple addresses for the same name.
+
+## Empty or undefined data
+Both empty and undefined data means that no data is expected. In the
+specification of the methods a destinction is made between the two. For the
+scenarios this distinction is upheld. In an empty set it will say either
+`(empty)` or `(undefined)`.
```

Comment:
The wording of this paragraph is awkward to me: the sentence “both empty and undefined data means that no data is expected”, thus suggesting that empty and undefined data are treated the same in scenarios – something I’d disagree with. But the next sentence contradicts that.

How about:
> In the methods’ specification, a distinction is made between empty sets, meaning the result is defined to be a collection of zero items, and undefined sets, meaning that no meaningful result can be provided. Empty sets are denoted as `(empty)` and undefined sets as `(undefined)`.